forte/data/vocabulary.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import List, Tuple, Dict, Union, Hashable, Iterable
+
+
+class Vocabulary:
+    '''This class maps element to representation. Element
+    could be any hashable type and there are two types of
+    representations, namely, "indexing" and "one-hot". There
+    are two special types of element, namely PAD_ELEMENT
+    and UNK_ELEMENT.
+    For "indexing" vocabulary,
+        Element:  <PAD>  ele1   ele2   ele3  ...
+        Id:         0      1      2      3   ...
+        Repr:       0      1      2      3   ...
+    For "one-hot" vocabulary,
+        Element:  <PAD>  ele1   ele2   ele3  ...
+        Id:        -1      0      1      2   ...
+        Repr:      [0,    [1,    [0,    [0,  ...
+                    0,     0,     1,     0,  ...
+                    0,     0,     0,     1,  ...
+                    0,     0,     0,     0,  ...
+                    ...]   ...]   ...]   ...]
+    If vocabulary uses UNK_ELEMENT, the first element,
+    "ele1" will be UNK_ELEMENT and any other elements
+    that cannot be found in the current vocabulary will
+    be mapped to the UNK_ELEMENT. Otherwise, UNK_ELEMENT
+    is not used. Error will occur when querying unknown
+    element in the vocabulary.
+    '''
+    PAD_ELEMENT = "<PAD>"
+    UNK_ELEMENT = "<UNK>"
+
+    def __init__(self, method: str, use_unk: bool):
+        self.element2id_dict = dict()
```

Comment:
Type annotation for these dict would be very helpful, people would know what to put here (and the editors can help too)

Reply:
Added.